File: account-service/src/main/java/com/vegesoft/moneytracker/account/client/AccountHistoryClient.java
package com.vegesoft.moneytracker.account.client;

import com.vegesoft.moneytracker.account.client.data.ExpenseRequest;
import com.vegesoft.moneytracker.account.client.data.IncomeRequest;
import java.util.UUID;
import reactor.core.publisher.Mono;

public interface AccountHistoryClient {

    Mono<UUID> expense(final ExpenseRequest expenseRequest);

    Mono<UUID> income(final IncomeRequest incomeRequest);
}

File: statistics-service/src/main/kotlin/com/vegesoft/moneytracker/statistics/client/AccountHistoryClient.kt
package com.vegesoft.moneytracker.statistics.client

import com.vegesoft.moneytracker.statistics.client.request.TransactionRequest
import com.vegesoft.moneytracker.statistics.client.response.TransactionResponse
import reactor.core.publisher.Flux

interface AccountHistoryClient {

    fun findTransactionViews(request: TransactionRequest): Flux<TransactionResponse>
}